### PR TITLE
Implement publishing new message for PAT repo selection

### PR DESCRIPTION
**Pull request by tapishr**

### CODE REVIEW

**Review by MuskanPaliwal — APPROVED** [review]
just a few comments, otherwise it looks good to me.

File: components/setup/RepoSelection.tsx [hunk 1]
```diff
@@ -93,23 +95,24 @@ const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { re
...
 					console.info(`[RepoSelection/handleSubmit] repos data saved successfully in db`);
-					if(setIsRepoSelectionDone) {setIsRepoSelectionDone(true) };
+					if (setIsRepoSelectionDone) { setIsRepoSelectionDone(true) };
+					if (isPAT) { setSubmitButtonText("Submitted") }
 				}
 			})
 			.catch((error) => {
 				setError(`Unable to submit selected repos, \nPlease refresh this page and try again.`);
-				if(setIsRepoSelectionDone) {setIsRepoSelectionDone(false) };
+				if (setIsRepoSelectionDone) { setIsRepoSelectionDone(false) };
```
[the remark this answers is not in the document]
Comment (tapishr) [reply components/setup/RepoSelection.tsx:111]: done

[the remark this answers is not in the document]
Comment (tapishr) [reply components/setup/RepoSelection.tsx]: had to revert this as well

File: components/setup/BuildInstruction.tsx [hunk 2]
```diff
@@ -126,7 +126,7 @@ const BuildInstruction: React.FC<BuildInstructionProps> = ({ selectedHosting, us
 			);
 		} else {
 			if (!isRepoSelectionDone) {
-				return (<RepoSelection repoProvider={selectedProvider} installId={installId} setIsRepoSelectionDone={setIsRepoSelectionDone} />)
+				return (<RepoSelection repoProvider={selectedProvider} installId={installId} setIsRepoSelectionDone={setIsRepoSelectionDone} isPAT={false} />)
```
[the remark this answers is not in the document]
Comment (tapishr) [reply components/setup/BuildInstruction.tsx:129]: changed to `isNewAccordion`, to tell the component if it is being rendered in a new accordion or the same one as docker instructions

File: components/setup/RepoSelection.tsx [hunk 3]
```diff
@@ -93,23 +95,24 @@ const RepoSelection = ({ repoProvider, installId, setIsRepoSelectionDone }: { re
 	const handleSubmit = () => {
 		setIsRepoSubmitButtonDisabled(true)
 		const reposListInSetupArgs = formatRepoListInSaveSetupArgsForm(selectedRepos, installId);
-		axios.post('/api/dpu/setup', { info: reposListInSetupArgs, installationId: installId })
+		axios.post('/api/dpu/setup', { info: reposListInSetupArgs, installationId: installId, isPublish: isPAT })
```
[the remark this answers is not in the document]
Comment (tapishr) [reply components/setup/RepoSelection.tsx:98]: `isPublish` is used in `/api/dpu/setup` to tell the api if they have to publish the message to pubsub (only true when provider === bitbucket && hosting === self)

**Review by MuskanPaliwal — DISMISSED** [review]

**Review by MuskanPaliwal — DISMISSED** [review]

**Review by MuskanPaliwal — APPROVED** [review]

File: components/setup/RepoSelection.tsx [hunk 4]
```diff
@@ -31,47 +31,69 @@ function formatRepoListInSaveSetupArgsForm(repos: RepoIdentifier[], install_id:
...
 				})
-				.then((repos) => {
-					const filteredRepos = providerReposForUser.filter(repo => repos.some(selectedRepo => 
-						selectedRepo.repo_provider === repo.repo_provider &&
-						selectedRepo.repo_owner === repo.repo_owner &&
-						selectedRepo.repo_name === repo.repo_name
-					)); // this is important because we are matching object references when we compare values in the checkbox
-					setSelectedRepos(filteredRepos);
+				.then((allRepos) => {
+					const providerReposForUser = allRepos.filter(repo => repo.repo_provider === repoProvider);
```
Comment (avikalpg) [inline components/setup/RepoSelection.tsx:53]: This seems redundant (given that the `axios.get` function call is sending the name of the provider in params.

**Pull request merged**